src/DataCollection.Shared/ViewModels/AttachmentsViewModel.cs

```diff
@@ -320,6 +332,44 @@ private async void AddNewAttachment(string filePath)
             }
         }
 
+        private string TruncateAndCopyFile(string filePath)
+        {
+            // Copy the attachment to a temp directory with a name less than 40 chars
+            // Do nothing if attachment name is already 40 or fewer chars
+
+            // Truncate early if name is longer than 40 chars
+            var filename = Path.GetFileNameWithoutExtension(filePath);
+
+            if (filename.Length > 40)
+            {
+                var filebase = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
+                    "AttachmentTemp", Guid.NewGuid().ToString());
+
+                if (!Directory.Exists(filebase))
+                {
+                    Directory.CreateDirectory(filebase);
+                }
+
+                filename = filename.Substring(0, 40);
```

Comment:
Should this be
`filename = filename.Substring(0, 40 - extension.Length);` ?

Reply:
You would think that, but it seems the 40 char limitation only applies to the filename, exclusive of the extension. I tried with a geodatabase file and was able to successfully upload with the full .geodatabase extension (name ended up being 52 chars with the extension).